explorer-api/src/Explorer/Web/Api/Legacy/BlocksTxs.hs

```diff
@@ -103,10 +103,9 @@ queryCTxBriefs xs = do
 queryTxInputs :: MonadIO m => [TxId] -> SqlPersistT m [(TxId, [CTxAddressBrief])]
 queryTxInputs txids = do
     rows <- select . distinct . from $ \(tx `InnerJoin` txIn `InnerJoin` txOut `InnerJoin` txInTx) -> do
-              on (txInTx ^. TxId ==. txIn ^. TxInTxOutId)
-              on (txIn ^. TxInTxOutId ==. txOut ^. TxOutTxId
-                  &&. txIn ^. TxInTxOutIndex ==. txOut ^. TxOutIndex)
               on (tx ^. TxId ==. txIn ^. TxInTxInId)
+              on (txIn ^. TxInTxOutId ==. txOut ^. TxOutTxId &&. txIn ^. TxInTxOutIndex ==. txOut ^. TxOutIndex)
+              on (txOut ^. TxOutTxId ==. txInTx ^. TxId)
```

Comment:
These fliters are exactly the same as `queryTxInputs` in the other file.
Actually the whole function looks fairly similar.
I suppose if this is legacy code it doesn't matter too much though.
It's not repeated any where else, is it?

Reply:
It actually is. There's _a lot_ of duplication in the SQL queries here and there and I never really took the time to address any of them.